### PR TITLE
DCKUBE-436 Bitbucket NFS server chart.

### PR DESCRIPTION
**Pull request by wkritzinger-atlassian**
TODO: 
* Fix docker image that's used for the tag
* Tune some of the variables for the nfs chart

### CODE REVIEW

File: src/test/infrastructure/nfs-server/values.yaml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+image:
+  repository: docker.atl-paas.net/atlassian/nfs-server-test
+  #pullPolicy: IfNotPresent
+  pullPolicy: Always
+  # Overrides the image tag whose default is the chart appVersion.
+  tag: "wkritzinger"
+
+imagePullSecrets:
+  - name: regcred
+nameOverride: ""
```
Comment (nanux) [inline src/test/infrastructure/nfs-server/values.yaml:16]: I believe you can leave out `""` for empty values.

**Review by tarka — APPROVED** [review]

**Review by nanux — APPROVED** [review]

File: src/test/infrastructure/nfs-server/values.yaml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+# Declare variables to be passed into your templates.
+
+replicaCount: 1
+
+image:
+  repository: docker.atl-paas.net/atlassian/nfs-server-test
+  #pullPolicy: IfNotPresent
+  pullPolicy: Always
+  # Overrides the image tag whose default is the chart appVersion.
+  tag: "wkritzinger"
```
[the remark this answers is not in the document]
Comment (wkritzinger-atlassian) [reply src/test/infrastructure/nfs-server/values.yaml:12]: Yep once the Docker image was merged and pushed I'll update this

File: src/test/infrastructure/nfs-server/values.yaml (hunk 1 above)
Comment (wkritzinger-atlassian) [reply src/test/infrastructure/nfs-server/values.yaml:16]: This came from the helm chart template, as did most of these values in this file so I might leave it as-is

File: pom.xml [hunk 3]
```diff
@@ -118,6 +118,14 @@
 
         <persistence.enabled>true</persistence.enabled>
         <clustering.enabled>true</clustering.enabled>
+        <!--
+            Number of instance replicas to start during initial setup. This value is currently only used by Bitbucket,
+            other products' values.yaml ignore this property. Other products need to perform additional setup steps on
+            a single node after Helm chart installation.
+        -->
+        <replica.count>2</replica.count>
```
Comment (wkritzinger-atlassian) [inline pom.xml:126]: I made this change so the setup for Bitbucket becomes easier. It does not scale up the instance after deployment anymore and instead starts up with 2 nodes. All other products still work the same way, this variable is only used for the Bitbucket templates, as is mentioned in the comment above.

File: src/test/config/bitbucket/values.yaml [hunk 4]
```diff
@@ -1,5 +1,7 @@
 # This file contains overrides for the Bitbucket Helm chart's values.yaml file
 
+replicaCount: ${replica.count}
```
Comment (wkritzinger-atlassian) [inline src/test/config/bitbucket/values.yaml:3]: This is how this value is only applied to Bitbucket, this line is missing in the other products.

**Comment by wkritzinger-atlassian** [comment]
Fixed the TODOs and got green builds using this branch. Need a new release before we get green builds again for Bitbucket.

**Pull request merged**